components/org.wso2.carbon.identity.api.user.functionality/org.wso2.carbon.identity.rest.api.user.functionality.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/functionality/v1/core/UserFunctionalityService.java

```diff
@@ -0,0 +1,322 @@
+                            .getFunctionalityLockCode())) {
+                throw handleError(Response.Status.FORBIDDEN,
+                        Constants.ErrorMessages.ERROR_CODE_FUNCTIONALITY_ALREADY_LOCKED);
+            }
+            userFunctionalityManager.lock(userId,
+                    tenantId, functionalityIdentifier, unlockTime,
+                    UserFunctionalityMgtConstants.FunctionalityLockReasons.USER_MANUALLY_LOCKED
+                            .getFunctionalityLockCode(),
+                    UserFunctionalityMgtConstants.FunctionalityLockReasons.USER_MANUALLY_LOCKED
+                            .getFunctionalityLockReason());
+            return Response.ok().build();
+        } catch (UserFunctionalityManagementException e) {
+            if (e instanceof UserFunctionalityManagementClientException) {
+                throw handleUserFunctionalityMgtClientException(e);
+            }
+            log.error("Error occurred while calling user functionality management service.", e);
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_LOCK_THE_FUNCTIONALITY_FAILED);
+        }
+    }
+
+    /**
+     * Get LockStatus of a functionality by a user.
+     *
+     * @param functionalityIdentifier FunctionalityIdentifier
+     * @return LockStatusResponse
+     */
+    public LockStatusResponse getLockStatusOfLoggedInUser(String functionalityIdentifier) {
+
+        try {
+            String userId = getUserIdFromUser(getUserFromContext());
+            LockStatusResponse lockStatusResponse;
+            int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+            UserFunctionalityManager userFunctionalityManager =
+                    UserFunctionalityServiceHolder.getuserFunctionalityManager();
+            FunctionalityLockStatus functionalityLockStatus = userFunctionalityManager
+                    .getLockStatus(userId, tenantId, functionalityIdentifier);
+            lockStatusResponse = new FunctionalityLockStatusToExternal().apply(functionalityLockStatus);
+            return lockStatusResponse;
+        } catch (UserFunctionalityManagementException e) {
+            if (e instanceof UserFunctionalityManagementClientException) {
+                throw handleUserFunctionalityMgtClientException(e);
+            }
+            log.error("Error occurred while calling user functionality management service.", e);
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    Constants.ErrorMessages.ERROR_CODE_GET_LOCK_STATUS_FAILED);
+        }
+    }
+
+    private APIError handleUserFunctionalityMgtClientException(UserFunctionalityManagementException e) {
+
+        if (StringUtils.equals(UserFunctionalityMgtConstants.ErrorMessages.USER_NOT_FOUND.getCode(),
+                e.getErrorCode())) {
+            log.debug("Cannot find the user corresponding to the userId.", e);
```

Comment:
should this be a debug log or an error log? if it's a debug log please check if debug logs are enabled first.

Reply:
check other instances as well

Reply:
It should be debug log. If it is error logs, then console will be flooded withlogs when a invalid request is sent by the user. User may try multiple times with invalid request and logs will be  flooded